src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1272,6 +1273,41 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return true;
     }
 
+    winrt::fire_and_forget TermControl::_restoreInBackground()
+    {
+        const auto path = std::exchange(_restorePath, {});
+
+        const auto weakSelf = get_weak();
+        co_await winrt::resume_background();
+        const auto self = weakSelf.get();
+
+        if (!self)
+        {
+            co_return;
+        }
+
+        const auto core = winrt::get_self<ControlCore>(_core);

```

Comment:
```suggestion
        const auto core = winrt::get_self<ControlCore>(self->_core);
```
Shouldn't we use `self` here?

Reply:
We don't need to. WinRT types like this are immovable and so the `this` pointer will refer to the same object after the suspension as it did before. `self` is just another safe/managed pointer.